src/api_module.js

```diff
@@ -8,194 +8,301 @@ const fetchRetry = useFetchRetry(fetch, {
+}
 
-    if (routes.includes("destroy")) {
-      this.destroy = async function (auth, id) {
-        const request = new Request(`${this.path}/${id}`, {
-          method: "DELETE",
-          headers: {
-            "x-secret": auth.secret,
-            "x-device-id": auth.device_id,
-          },
-        });
-        return await this._resolveRequest(request);
-      };
-    }
+async function httpPost(path, auth, data) {
+  const request = new Request(path, {
+    method: "POST",
+    headers: {
+      "content-type": "application/json",
+      "x-secret": auth.secret,
+      "x-device-id": auth.device_id,
+    },
+    body: JSON.stringify(data),
+  });
+  return await resolve(request);
+}
 
-    if (routes.includes("destroyEmpty")) {
-      this.destroyEmpty = async function (auth) {
-        const request = new Request(`${this.path}/destroy_empty`, {
-          method: "POST",
-          headers: {
-            "content-type": "application/json",
-            "x-secret": auth.secret,
-            "x-device-id": auth.device_id,
-          },
-        });
-        return await this._resolveRequest(request);
-      };
-    }
+async function httpPatch(path, auth, data) {
+  const request = new Request(path, {
+    method: "PATCH",
+    headers: {
+      "content-type": "application/json",
+      "x-secret": auth.secret,
+      "x-device-id": auth.device_id,
+    },
+    body: JSON.stringify(data),
+  });
+  return await resolve(request);
+}
 
-    if (routes.includes("merge")) {
-      this.merge = async function (auth, newID, oldID) {
-        const request = new Request(
-          `${this.path}/${newID}/merge?source_id=${oldID}`,
-          {
-            method: "POST",
-            headers: {
-              "content-type": "application/json",
-              "x-secret": auth.secret,
-              "x-device-id": auth.device_id,
-            },
-          }
-        );
-        return await this._resolveRequest(request);
-      };
-    }
+async function httpDelete(path, auth) {
+  const request = new Request(path, {
+    method: "DELETE",
+    headers: {
+      "x-secret": auth.secret,
+      "x-device-id": auth.device_id,
+    },
+  });
+  return await resolve(request);
+}
+
+async function resolve(request) {
+  let response, result;
+  try {
+    response = await fetchRetry(request);
+    result = response.status === 204 ? true : await response.json();
+  } catch (error) {
+    const reason = {};
+    reason[error.constructor.name] = [error.message];
+    throw reason;
   }
+  if (response.ok) {
+    return result;
+  } else {
+    throw result;
+  }
+}
 
-  async _resolveRequest(request) {
-    let response, result;
-    try {
-      response = await fetchRetry(request);
-      result = response.status === 204 ? true : await response.json();
-    } catch (reason) {
-      throw { error: [reason] };
-    }
-    if (response.ok) {
-      return result;
-    } else {
-      throw result;
-    }
+class CRDModule {
```

Comment:
I would maybe create a `BaseModule` and use [composition](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#mix-ins) but can live with this structure